### ISSUE TITLE
PDF reader: Ability to hide some or all annotations

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#2391]
https://forums.zotero.org/discussion/comment/400691/#Comment_400691

Hide all and/or hide by color

The easiest would be for the PDF view to update to reflect the filtered annotations in the sidebar. Any reason not to do that? It would be roughly equivalent to hiding items that don't match a given selected tag in the library view.

Not sure how you'd hide all, though — that would need to be a different mechanism. Maybe just a View setting?

**Comment by mrtcode** [comment]
Yes, I think we can do both things. View setting would be persistent?

**Comment by dstillman** [comment]
Persistent as in if you switched to a different PDF would the annotations be hidden? I think so. The alternative seems too complicated.

**Comment by mrtcode** [comment]
Also I was thinking if we want to save this together with view settings in `.zotero-pdf-state`?

**Comment by dstillman** [comment]
Oh, I guess we could do that. Remind me what other settings we store in there? I'm not sure if this makes more sense as global setting or an item-specific one.

**Comment by mrtcode** [comment]
We are storing various "view" parameters:

```json
{
  "pageIndex":5,
  "scale":"page-width",
  "rotation":0,
  "top":24,
  "left":-14,
  "sidebarView":9,
  "sidebarWidth":240,
  "scrollMode":0,
  "spreadMode":0
}
```

**Comment by mrtcode** [comment]
And if we search annotations in the sidebar, do we want to hide unmatched annotation in pages view as well?

**Comment by mrtcode** [comment]
> Not sure how you'd hide all, though — that would need to be a different mechanism. Maybe just a View setting?

So if we completely hide all annotations (with a View setting) that means we should disable annotating capabilities? Because you won't see what you are creating. Or that should show all annotations again?

**Comment by dstillman** [comment]
Should show all again. Similar to (though a little weirder than) creating a new item with an active quick search in the items list.

**Comment by mrtcode** [comment]
Do we want this setting to only have an effect for PDF view and still keep annotations visible in the sidebar? If so, it would make sense to have that view setting in PDF page context menu. This way you can quickly hide annotations when you just want to cleanly read the document.

**Comment by dstillman** [comment]
Yeah, I guess that makes sense. Because what's the alternative? An empty annotations sidebar? So then, yeah, showing this in the View menu and in the page context menu seems right.

**Comment by dstillman** [comment]
But this feels more like a global setting of the PDF reader than a property of the PDF to me. Seems a little weird/annoying to have to fiddle with this for each PDF.

**Comment by mrtcode** [comment]
Ok, we can implement this as a global setting for now. Should we persist it? And where, if so?

**Comment by dstillman** [comment]
Just a pref?

**Comment by dstillman** [comment]
Unless we can reuse any of the persistence stuff we use for panes in the library view?

**Comment by mrtcode** [comment]
Well, we're doing [this](https://github.com/zotero/zotero/blob/66fccec6b454fc0b6be622e1b8193c2febf1d7f3/chrome/content/zotero/zoteroPane.xul#L79-L80), but I'm not sure what is better in this case.

**Comment by dstillman** [comment]
Yeah, I think just a pref makes sense here.

**Comment by mrtcode** [comment]
So it will be "Show Annotations" option after "Split Vertically" in page context and View menus. Or, alternatively, it could be the first option.

**Comment by mrtcode** [comment]
And the global pref will be named `zotero.reader.showAnnotations`.

**Comment by mrtcode** [comment]
Should it enable showing annotations if you click an annotation in the sidebar?

**Comment by mrtcode** [comment]
Do we want individual "Show Annotations" controls for primary and second views (as we are leaning towards feature complete second view in split view mode now)?

**Comment by dstillman** [comment]
> Should it enable showing annotations if you click an annotation in the sidebar?

I don't think so. You might want to interact with your annotations solely in the sidebar without cluttering the PDF view?

> Do we want individual "Show Annotations" controls for primary and second views (as we are leaning towards feature complete second view in split view mode now)?

Eh, no. Global pref.

**Comment by mfranzil** [comment]
Hi. Any news on this feature? Would be a great addition IMO.

**Comment by mrtcode** [comment]
https://forums.zotero.org/discussion/127837/how-to-hide-highlight
